packages/babel-plugin-material-ui/src/index.ts

```diff
@@ -0,0 +1,117 @@
+import * as babel from '@babel/core';
+import * as t from '@babel/types';
+import resolveImports, { MappedImportsType, ImportType } from './resolveImports';
+import { Packages } from './packages';
+
+/**
+ * Holds the import name and location of everything found in the index file of each package
+ */
+const importMaps: { [K: string]: MappedImportsType | undefined } = {};
+
+/**
+ * Used to transform the specified root import to specific imports based on the exports found
+ * in the index file of the package
+ */
+function transformImport(path: babel.NodePath<t.ImportDeclaration>, packageName: string) {
+  if (importMaps[packageName] === undefined) {
+    importMaps[packageName] = resolveImports(packageName);
+  }
+
+  const { node } = path;
+  const importMap = importMaps[packageName]!;
+
+  for (const key in importMap) {
+    if (!importMap.hasOwnProperty(key)) {
+      continue;
+    }
+
+    const elements = importMap[key];
+    const newSpecifiers: Array<
+      t.ImportDefaultSpecifier | t.ImportSpecifier | t.ImportNamespaceSpecifier
+    > = [];
+
+    node.specifiers = node.specifiers.filter(spec => {
+      if (!t.isImportSpecifier(spec)) {
+        return true;
+      }
+
+      const mappedImport = elements.find(x => x.name === spec.imported.name);
+      if (!mappedImport) {
+        return true;
+      }
+
+      newSpecifiers.push(
+        mappedImport.type === ImportType.Namespace
```

Comment:
Babel has extra type helpers for this. No need to use TS enums.

Reply:
Could you show me what you mean please

Reply:
There's not documented API as far as I know only types so I would have to guess. It should be something like `babel.types.isImportNamespaceSpecifier`

Reply:
Ah, I see. `mappedImport` isn't a babel node so that won't be of any use here